lib/compress.js

```diff
@@ -3071,8 +3071,13 @@ merge(Compressor.prototype, {
         // If the node has been successfully reduced to a constant,
         // then its value is returned; otherwise the element itself
         // is returned.
+        //
         // They can be distinguished as constant value is never a
         // descendant of AST_Node.
+        //
+        // When `ignore_side_effects` is `true`, inspect the constant value
+        // produced without worrying about any side effects caused by said
+        // expression.
```

Comment:
@kzc as per suggestion in https://github.com/mishoo/UglifyJS2/pull/3714#issuecomment-584964201